### ISSUE TITLE
Failed to create WDA session,  Original error: +[XCAXClient_iOS sharedClient]: unrecognized selector sent to class 0x102d48130

### ISSUE THREAD

**Issue by AppiumIOSRuser** [appium/appium#12462]
## The problem
Unable to create WDA session to launch URL in Safari browser, after upgrading to Xcode10.2 

## Environment

* Appium version (or git revision) that exhibits the issue:1.12.1
* Last Appium version that did not exhibit the issue (if applicable):1.8.2 beta
* Desktop OS/version used to run Appium: Mac OS Mojave 10.14.4
* Node.js version (unless using Appium.app|exe): v11.6.0
* Npm or Yarn package manager:6.5.0
* Mobile platform/version under test: ios 12.2
* Real device or emulator/simulator:iPhoneX real device
* Xcode Version: 10.2
* Appium CLI or Appium.app|exe:

## Details

The setup used to work with xcode 10.1 and ios 12.1 and after upgrading ios version to 12.2 and xcode version to 10.2 the script fail to launch the url in safari browser and  getting the below error 

2019-04-09 14:08:12:082 [WD Proxy] Got response with status 200: {"value":"+[XCAXClient_iOS sharedClient]: unrecognized selector sent to class 0x102d48130\n\n(\n\t0   CoreFoundation                      0x00000001e24f8530 <redacted> + 252\n\t1   libobjc.A.dylib                     0x00000001e16d39f8 objc_exception_throw + 56\n\t2   CoreFoundation                      0x00000001e241a2c4 <redacted> + 0\n\t3   CoreFoundation                      0x00000001e24fdd60 <redacted> + 1408\n\t4   CoreFoundation                      0x00000001e24ff9fc _CF_forwarding_prep_0 + 92\n\t5   WebDriverAgentLib                   0x0000000104bdc3f0 __37+[FBApplication fb_activeApplication]_block_invoke + 52\n\t6   WebDriverAgentLib                   0x0000000104bb5520 -[FBRunLoopSpinner spinUntilTrue:error:] + 152\n\t7   WebDriverAgentLib                   0x0000000104bb5460 -[FBRunLoopSpinner spinUntilTrue:] + 84\n\t8   WebDriverAgentLib                   0x0000000104bdc150 +[FBApplication fb_activeApplication] + 128\n\t9   WebDriverAgentLib                   0x0000000104bc7b94 -[FBSession ...

2019-04-09 14:08:12:083 [WD Proxy] Got an unexpected response: {"value":"+[XCAXClient_iOS sharedClient]: unrecognized selector sent to class 0x102d48130\n\n(\n\t0   CoreFoundation                      0x00000001e24f8530 <redacted> + 252\n\t1   libobjc.A.dylib                     0x00000001e16d39f8 objc_exception_throw + 56\n\t2   CoreFoundation              ...

2019-04-09 14:08:12:083 [MJSONWP] Matched JSONWP error code 13 to UnknownError

## Link to Appium logs

 [AppiumLog](https://gist.github.com/AppiumIOSRuser/c29d108e300ecc1232856bd40e335f0a)

**Comment by mykola-mokhnach** [comment]
Make sure the WDA binary on the device is up to date. The simplest way to verify that would be to delete the binary manually and clear the Xcode caches before starting a new test

**Comment by AppiumIOSRuser** [comment]
I have deleted WDA binary from the device and also performed cleanup in the xcode by doing the following 
1. Clean the Build (Command+Shift+K)
2. Clean the Build Folder(Command+Option+Shift+K)
3. Deleted the  contents of Derived Data  folder

And I am still getting the same error I have attached the Logs for the execution happen after the cleanup
[log.txt](https://github.com/appium/appium/files/3062612/log.txt)

**Comment by mykola-mokhnach** [comment]
Can you make sure that the WDA source at /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent is similar to what is stored at https://github.com/appium/WebDriverAgent ?

**Comment by AppiumIOSRuser** [comment]
Hi @mykola-mokhnach ,
I have already downloaded the latest  WDA  from git and tried yesterday and it haven't helped.

**Comment by mykola-mokhnach** [comment]
I'm out of ideas. Check if the stuff works if you use a Simulator instead of a real device, try to run any integration test from Xcode...

**Comment by 1312818364** [comment]
https://github.com/facebook/WebDriverAgent/issues/1076
here can resolve your problem

**Comment by AppiumIOSRuser** [comment]
Hi @mykola-mokhnach,
I have executed the same test in simulator and getting the same error.I am attaching the log below.
[log.txt](https://github.com/appium/appium/files/3072203/log.txt)

And Also executed the integration Test in Xcode please find the log below
[Xcode Integration Test.txt](https://github.com/appium/appium/files/3072302/Xcode.Integration.Test.txt)

**Comment by AppiumIOSRuser** [comment]
Hi @mykola-mokhnach & @1312818364 ,
Thanks for spending time and looking into this issue.
I have gone through facebook/WebDriverAgent#1076  and I understood that I need to wait for fix in WDA project. Please correct me if my understating is wrong.

**Comment by AppiumIOSRuser** [comment]
Hi @mykola-mokhnach & @1312818364 ,
Any update on this please. Kindly please confirm if I have to wait for the fix in WDA project or is there any workaround available ?

**Comment by 1312818364** [comment]
you just need download the lastest Appium WebDriverAgent, not Facebook WebDriverAgent.

**Comment by 1312818364** [comment]
@AppiumIOSRuser 
 you just need download the lastest Appium WebDriverAgent, not Facebook WebDriverAgent,

 git clone  this https://github.com/appium/WebDriverAgent.git,  and run ./Script/bootstrap.sh ,  this WebDriverAgent implements the   [XCAXClient_iOS sharedClient] method.

**Comment by 1312818364** [comment]
![image](https://user-images.githubusercontent.com/7345419/56197970-8542e900-606c-11e9-8d52-3b2e027680aa.png)

**Comment by AppiumIOSRuser** [comment]
@1312818364  & @mykola-mokhnach 
 I have downloaded the appium  WDA project Now I am not getting error related to XCAXClient_iOS sharedClient. But the url is still not getting launched in safari browser.Please check the below log

Log: https://gist.github.com/AppiumIOSRuser/cbc68c48d4d9bd2286348e31cad2be75

Appium version :1.12.1
Xcode Version : 10.2

**Comment by 1312818364** [comment]
@AppiumIOSRuser  你需要修改新下载的appium WDA的  签名和product bundle ID，删除手机上之前安装的WebdriverAgent, 然后重新build 安装WebDriverAgent到你的手机上，之后连接手机command + U 运行WebDriverAgentRunner 测试用例，保证其处于Testing状态。 现在的错误告诉你： 你的WDA在手机上没有运行。

**Comment by mykola-mokhnach** [comment]
> But the url is still not getting launched in safari browser.Please check the below log

This is another issue and it is already in progress, because Apple has changed quite a lot of things in Xcode 10.2 SDK. 

Closing as not a problem

**Comment by AppiumIOSRuser** [comment]
After upgrading the Appium to v1.13.0-beta.3 and ios-webkit-debug-proxy to v1.8.4.I am able to execute test in Safari. 

@1312818364 & @mykola-mokhnach,
Thanks for your support.

**Comment by aaronmagi** [comment]
@AppiumIOSRuser  where do you get the beta version of Appium desktop?  I ran into this issue today.

Thanks

**Comment by LiebeKrake** [comment]
> @AppiumIOSRuser 你需要修改新下载的appium WDA的 签名和product bundle ID，删除手机上之前安装的WebdriverAgent, 然后重新build 安装WebDriverAgent到你的手机上，之后连接手机command + U 运行WebDriverAgentRunner 测试用例，保证其处于Testing状态。 现在的错误告诉你： 你的WDA在手机上没有运行。

Hello~ 
我按照你之前推荐的步骤：
clone了 Appium的WebDriverAgent，运行 ./Script/bootstrap.sh ，报“cannot find module eslint-config-appium”；
然后我安装了eslint-config-appium，再运行 ./Script/bootstrap.sh ，又报“cannot find module eslint-plugin-import”；
之后我又安装了eslint-plugin-import，再运行 ./Script/bootstrap.sh ，又报“Cannot find module 'eslint/lib/util/source-code”

`Cannot find module 'eslint/lib/util/source-code'
Referenced from: /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/.eslintrc
Error: Cannot find module 'eslint/lib/util/source-code'
Referenced from: /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/.eslintrc
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/eslint-plugin-import/lib/ExportMap.js:20:19)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
`
还有解吗？

我本地的环境是：
macOS：10.14.4
appium：1.9.1
nodeJS：10.15.2
npm：6.4.1

——————————————————————————————————————————————
I had git clone webdriveragent from [ https://github.com/appium/WebDriverAgent.git](url), and when i ran  ./Script/bootstrap.sh, it threw “cannot find module eslint-config-appium”；
After installed  eslint-config-appium, and ran  ./Script/bootstrap.sh, it threw "cannot find module eslint-plugin-import";
Then installed  eslint-plugin-import,  and ran  ./Script/bootstrap.sh, it threw "Cannot find module 'eslint/lib/util/source-code"

``Cannot find module 'eslint/lib/util/source-code'
Referenced from: /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/.eslintrc
Error: Cannot find module 'eslint/lib/util/source-code'
Referenced from: /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/.eslintrc
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/eslint-plugin-import/lib/ExportMap.js:20:19)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
``

environment：
macOS：10.14.4
appium：1.9.1
nodeJS：10.15.2
npm：6.4.1

I don't know how to solve it, please help me.

**Comment by AppiumIOSRuser** [comment]
@aaronmagi ,
I am not using Appium desktop. I am using appium server through code(http://www.automationtestinghub.com/3-ways-to-start-appium-server-from-java/)

Working version for me is Appium  v1.13.0-beta.3

**Comment by AppiumIOSRuser** [comment]
@LiebeKrake ,
Please try to run the script like this **./Scripts/bootstrap.sh -d** 
if you are using xcode 10.2 please upgrade your appium version. For me the working versions are 
Appium **v1.13.0-beta.3**
Xcode **10.2**
ios **12.2**
ios-webkit-debug-proxy  **v1.8.4**.

**Comment by LiebeKrake** [comment]
> @LiebeKrake ,
> Please try to run the script like this **./Scripts/bootstrap.sh -d**
> if you are using xcode 10.2 please upgrade your appium version. For me the working versions are
> Appium **v1.13.0-beta.3**
> Xcode **10.2**
> ios **12.2**
> ios-webkit-debug-proxy **v1.8.4**.

Thinks a lot, I'll try~

**Comment by galaxyling** [comment]
Has the problem been solved?I also encountered this problem, how to solve?

**Comment by sunshinelulu** [comment]
@AppiumIOSRuser 
after download  https://github.com/appium/WebDriverAgent.git, when i run “./Scripts/bootstrap.sh”，it tell me：
![image](https://user-images.githubusercontent.com/12216518/57592222-2a67b900-7568-11e9-862e-57b1c92fac13.png)

./Scripts/bootstrap.sh 
Fetching dependencies
*** Checking out RoutingHTTPServer at "v1.0.2"
*** Checking out YYCache at "1.1.0"
*** Checking out CocoaAsyncSocket at "7.6.3"
*** xcodebuild output can be found in /var/folders/ng/qqhfgsv933335vxc_66919rr0000gp/T/carthage-xcodebuild.V1Zube.log
*** Downloading CocoaAsyncSocket.framework binary at "Version 7.6.3"
*** Building scheme "RoutingHTTPServer tvOS" in RoutingHTTPServer.xcodeproj
Could not find any available simulators for tvOS

**Comment by KazuCocoa** [comment]
What about updating your carthage version?

**Comment by Samira8106** [comment]
@LiebeKrake  which nodejs version are you used?
